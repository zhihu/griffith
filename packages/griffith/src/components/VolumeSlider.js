import React, {Component} from 'react'
import {css} from 'aphrodite/no-important'
import Slider from './Slider'
import styles, {slider as sliderStyles} from './VolumeSlider.styles'

class VolumeSlider extends Component {
  render() {
    return (
      <div className={css(styles.root)}>
        <Slider
          {...this.props}
          styles={sliderStyles}
          orientation="vertical"
          step={0.1}
        />
      </div>
    )
  }
}

export default VolumeSlider
