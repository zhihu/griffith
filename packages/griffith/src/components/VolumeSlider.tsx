import React, {Component} from 'react'
import {css} from 'aphrodite/no-important'
import Slider from './Slider'
import styles, {slider as sliderStyles} from './VolumeSlider.styles'

class VolumeSlider extends Component {
  render() {
    return (
      <div className={css(styles.root)}>
        <Slider
          {...this.props}
          // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
          styles={sliderStyles}
          orientation="vertical"
          step={0.1}
        />
      </div>
    )
  }
}

export default VolumeSlider
