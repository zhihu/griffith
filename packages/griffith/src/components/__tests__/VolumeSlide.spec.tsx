import React from 'react'
import Renderer from 'react-test-renderer'
import VolumeSlider from '../VolumeSlider'

describe('VolumeSlider', () => {
  it('get VolumeSlider component', () => {
    expect(
      Renderer.create(<VolumeSlider total={1} value={0.5} />).toJSON()
    ).toMatchSnapshot()
  })
})
