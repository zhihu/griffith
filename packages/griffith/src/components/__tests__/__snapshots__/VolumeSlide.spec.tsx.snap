// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolumeSlider get VolumeSlider component 1`] = `
<div
  className="root_1cc9rz"
>
  <div
    className="root_17bkmwy-o_O-root_1bq0ydu-o_O-root_av3p6z"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="slider"
    tabIndex={0}
  >
    <div
      className="inner_9gm5i8-o_O-inner_1jtvojh"
    >
      <div
        className="track_1bopucd-o_O-track_31rx9n-o_O-track_ytl6g2"
      >
        <div
          className="bar_1kpj1ed-o_O-bar_1ql2ook"
          style={
            Object {
              "height": "100%",
              "transform": "scaleY(0.5)",
              "transformOrigin": "bottom",
            }
          }
        />
      </div>
      <div
        className="thumbWrapper_ay4wjb-o_O-thumbWrapper_drkmbm"
        style={
          Object {
            "height": "100%",
            "transform": "translateY(50%)",
            "transformOrigin": "bottom",
          }
        }
      >
        <div
          className="thumb_ozdqky-o_O-thumb_8gf5gg"
        />
      </div>
    </div>
  </div>
</div>
`;
